### ISSUE TITLE
php heredoc highlighting breaks

### ISSUE THREAD

**Issue by adamjimenez** [ajaxorg/ace#3171]
Heredoc highlighting with no spaces breaks e.g:

```
<?php
$bar = <<<EOD
highlightingWithNoSpaces
breaks
EOD;

$bar = <<<EOD
highlighting with spaces
doesn't break
EOD;
?>
```
![image](https://cloud.githubusercontent.com/assets/573192/21356439/8332dde4-c6c9-11e6-8ce0-ecc594338fa8.png)

**Comment by thirdwheel** [comment]
Experiencing a similar issue with nowdocs - using it to store SQL breaks highlighting on a BEGIN...END block.

```
<?php

$data=<<<'EOF'
DECLARE @CategoryID int = CAST(? as varchar), @EmployeeID nvarchar(75) = ?
DECLARE @Check TABLE
(
	CategoryID int,
	EmployeeID nvarchar(75)
)

IF @CategoryID = 0 
	DELETE Employee_Category WHERE Employee_Category_HRM_ID = @EmployeeID
ELSE
BEGIN
    UPDATE Employee_Category
    SET Employee_Category_ID = @CategoryID
    OUTPUT inserted.* INTO @Check
    WHERE Employee_Category_HRM_ID = @EmployeeID

    IF (SELECT COUNT(*) FROM @Check) = 0
        INSERT INTO Employee_Category
        VALUES(@CategoryID, @EmployeeID)
END

EOF;

echo $data;

?>
```
![ace clould9 php highlighting issue](https://user-images.githubusercontent.com/160601/30567938-01b248e4-9d16-11e7-9f45-e2e6825d98d5.PNG)

**Comment by thirdwheel** [comment]
This bug is still current

**Comment by thirdwheel** [comment]
![image](https://user-images.githubusercontent.com/160601/167510510-76964eb9-f1a3-42c1-a948-13ffa5a07fa2.png)

**Comment by barrydit** [comment]
I also wanted to note that using a HEREDOC in place of a parameter in function, also breaks highlighting. I used mine inside define('...', <<<HEREDOC
....
HEREDOC);

![image](https://user-images.githubusercontent.com/6217010/173412873-ef9d7137-e9ca-47c8-bdb0-9bc9270ebd33.png)

**Comment by andrewnester** [comment]
The change has been merged and will be released in upcoming version of Ace

**Comment by andrewnester** [comment]
Released in 1.8.0
